src/test/nodestore/DatabaseShard_test.cpp

```diff
@@ -0,0 +1,986 @@
+                    do
+                    {
+                        from = rand_int(rng_, n - 1);
+                        to = rand_int(rng_, n - 1);
+                    } while (from == to);
+
+                    pay.push_back(std::make_pair(from, to));
+                }
+
+                n += !rand_int(rng_, (ledgersPerShard * nShards) / dataSize);
+
+                if (n > A_.size())
+                {
+                    char str[9];
+                    for (int j = 0; j < 8; ++j)
+                        str[j] = 'a' + rand_int(rng_, 'z' - 'a');
+                    str[8] = 0;
+                    A_.push_back(std::make_shared<test::jtx::Account>(str));
+                }
+
+                acc_.push_back(n);
+                pay_.push_back(pay);
+                xrp_.push_back(rand_int(rng_, 90) + 10);
+            }
+        }
+
+        bool
+        newAcc(int seq)
+        {
+            return acc_[seq] > (seq ? acc_[seq - 1] : 0);
+        }
+
+        void
+        makeLedgerData(test::jtx::Env& env_, std::uint32_t seq)
+        {
+            using namespace test::jtx;
+
+            if (newAcc(seq))
+                env_.fund(XRP(iniAmount), *A_[acc_[seq] - 1]);
+
+            for (std::uint32_t i = 0; i < pay_[seq].size(); ++i)
+            {
+                env_(
+                    pay(*A_[pay_[seq][i].first],
+                        *A_[pay_[seq][i].second],
+                        XRP(xrp_[seq])));
+            }
+        }
+
+        bool
+        makeLedgers(test::jtx::Env& env_)
+        {
+            for (std::uint32_t i = 3; i <= ledgersPerShard; ++i)
+            {
+                if (!env_.close())
+                    return false;
+                std::shared_ptr<const Ledger> ledger =
+                    env_.app().getLedgerMaster().getClosedLedger();
+                if (ledger->info().seq != i)
+                    return false;
+            }
+
+            for (std::uint32_t i = 0; i < ledgersPerShard * nShards_; ++i)
+            {
+                makeLedgerData(env_, i);
+                if (!env_.close())
+                    return false;
+                std::shared_ptr<const Ledger> ledger =
+                    env_.app().getLedgerMaster().getClosedLedger();
+                if (ledger->info().seq != i + ledgersPerShard + 1)
+                    return false;
+                ledgers_.push_back(ledger);
+            }
+
+            return true;
+        }
+    };
+
+    void
+    testLedgerData(testData& D, std::shared_ptr<Ledger> l, std::uint32_t seq)
+    {
+        using namespace test::jtx;
+
+        auto rootCount{0};
+        auto accCount{0};
+        auto sothCount{0};
+        for (auto const& s : l->sles)
+        {
+            if (s->getType() == ltACCOUNT_ROOT)
+            {
+                int sq = s->getFieldU32(sfSequence);
+                int reqsq = -1;
+                const auto id = s->getAccountID(sfAccount);
+                int i;
```

Comment:
I'd put this declaration in the loop. I.e. `for (int i=0;...`

Reply:
Done